src/app/onboarding/basic-information/page.tsx

```diff
@@ -1,95 +1,262 @@
+  const formValues = form.watch();
+  const isEmpty = useMemo(
+    () =>
+      !(
+        formValues.firstName &&
+        formValues.lastName &&
+        formValues.city &&
+        formValues.phoneNumber &&
+        (formValues.canReads.length > 0 || formValues.canSpeaks.length > 0)
+      ),
+    [formValues],
+  );
 
-  useEffect(() => {
-    setFirstName(onboarding?.userProfile.first_name || '');
-    setLastName(onboarding?.userProfile.last_name || '');
-    if (
-      onboarding?.userProfile.first_name !== '' &&
-      onboarding?.userProfile.last_name !== '' &&
-      onboarding?.userProfile.location !== '' &&
-      onboarding?.canReads.size !== 0 &&
-      onboarding?.canSpeaks.size !== 0
-    ) {
-      onboarding?.setCanContinue(true);
-    } else {
-      onboarding?.setProgress(1);
-      onboarding?.setCanContinue(false);
-    }
-  }, [onboarding]);
+  // handle valid form submission
+  // - validity should be handled by Zod
+  const onSubmit = () => {
+    push(`/onboarding/${onboarding.flow[2].url}`);
+  };
 
   return (
-    <>
-      <H1>Basic Information</H1>
-      <LineDiv>
-        <TextInput
-          label="First Name"
-          placeholder="Jane"
-          type="firstName"
-          setValue={setFirstName}
-          value={firstName}
-          onChange={v => {
-            onboarding?.updateProfile({
-              first_name: v,
-            });
-          }}
-        />
-        <TextInput
-          label="Last Name"
-          placeholder="Doe"
-          type="lastName"
-          setValue={setLastName}
-          value={lastName}
-          onChange={v => {
-            onboarding?.updateProfile({
-              last_name: v,
-            });
-          }}
-        />
-      </LineDiv>
-      <BigDataDropdown
-        label="City"
-        options={cities}
-        onChange={v => {
-          onboarding?.updateProfile({
-            location: v || '',
-          });
-        }}
-        defaultValue={onboarding?.userProfile.location}
-        placeholder="Select a city"
-      />
-      <BigDataDropdown
-        label="What languages can you speak and understand?"
-        options={languages}
-        onChange={v => {
-          onboarding?.setCanSpeaks(v || new Set<string>());
-        }}
-        defaultValue={onboarding?.canSpeaks}
-        placeholder="Select a language"
-        multi
-      />
-      <BigDataDropdown
-        label="What languages can you read and write?"
-        options={languages}
-        onChange={v => {
-          onboarding?.setCanReads(v || new Set<string>());
-        }}
-        defaultValue={onboarding?.canReads}
-        placeholder="Select a language"
-        multi
-      />
-    </>
+    <FormProvider {...form}>
+      <CardForm onSubmit={form.handleSubmit(onSubmit)}>
+        <Styles.BackLink href={backlinkHref}>
+          <Icon type="leftArrow" />
+        </Styles.BackLink>
+
+        <H1Centered>Basic Information</H1Centered>
```

Comment:
On figma, there is a "* Required Fields" textbook on the top right on all onboarding pages. Are we planning to include that on all onboarding pages? 
<img width="432" alt="Screen Shot 2024-04-13 at 4 44 56 PM" src="https://github.com/calblueprint/immigration-justice-project/assets/64336644/eb30d937-c862-4941-b291-91030761e1cb">

Reply:
oh nice catch, i missed that